lib/rexml/parsers/baseparser.rb

```diff
@@ -349,8 +349,9 @@ def pull_event
         end
         begin
           @source.read if @source.buffer.size<2
-          if @source.buffer[0] == ?<
-            if @source.buffer[1] == ?/
+          pattern = @source.buffer[0,2]
```

Comment:
We may want to reduce one more `@source.buffer` call. 

```ruby
pattern = @source.buffer
if pattern.size < 2
  @source.read
  pattern = @source.buffer
end
```

Reply:
It worked with this content.
Thank you!